exercises/conor_finn/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,167 @@
+#include "main.h"
+#include <algorithm>
+#include <functional>
+
+using std::cin;
+using std::cout;
+using std::endl;
+using std::ifstream;
+using std::map;
+using std::ofstream;
+using std::pair;
+using std::string;
+using std::vector;
+
+/**
+ * @brief Creates a results file containing the number of
+ * counts of words of >4 characters in a user defined file.
+ *
+ * @return int 0 on success, >0 otherwise.
+ */
+int wordCounterInterface() {
+  // Number of characters a word must have to be counted.
+  const int characters = 5;
+  // Ask for a file path.
+  cout << "Please enter the file path:";
+  string path;
+  cin >> path;
+  ifstream filein;
+  filein.open(path.c_str());
+  // Create a file in the calling directory.
+  ofstream fileout;
+
+  map<string, int> counts; // Map to hold the counts of each word.
+
+  // Check if the file exists.
+  if (filein.is_open()) {
+    map<string, int> counts = countWords(filein, characters);
+    vector<pair<int, string>> sorted{sortMap(counts)};
+    cout << "Please enter the output path (paths that do not reach existing "
+            "files will create new ones):";
+    cin >> path;
+    fileout.open(path.c_str());
+    // Write headings to the file.
+    string::size_type maxLength = findMaxSize(sorted);
+    fileout << "Word" + string(maxLength - 4, ' ') + "\tCounts" << endl;
```

Comment:
We haven't checked if `fileout` is open before trying to write to it. If I enter an invalid file name, the program exits without error but I have no file!